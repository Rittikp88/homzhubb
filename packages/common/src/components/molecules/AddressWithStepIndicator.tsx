import React from 'react';
import { FlatList, StyleProp, StyleSheet, View, ViewStyle } from 'react-native';
import { useTranslation } from 'react-i18next';
import { PlatformUtils } from '@homzhub/common/src/utils/PlatformUtils';
import { useDown, useViewPort } from '@homzhub/common/src/utils/MediaQueryUtils';
import { IBadgeInfo } from '@homzhub/mobile/src/navigation/interfaces';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { Badge } from '@homzhub/common/src/components/atoms/Badge';
import { Label } from '@homzhub/common/src/components/atoms/Text';
import { ITypographyProps } from '@homzhub/common/src/components/atoms/Typography';
import { PropertyAddressCountry } from '@homzhub/common/src/components/molecules/PropertyAddressCountry';
import { IAmenitiesData, PropertyAmenities } from '@homzhub/common/src/components/molecules/PropertyAmenities';
import { TypeOfPlan } from '@homzhub/common/src/domain/models/AssetPlan';
import { ILabelColor } from '@homzhub/common/src/domain/models/LeaseTransaction';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { deviceBreakpoint } from '@homzhub/common/src/constants/DeviceBreakpoints';

interface IProps {
  steps: string[];
  currentIndex: number;
  isStepDone: boolean[];
  primaryAddress: string;
  subAddress: string;
  countryFlag: React.ReactElement;
  propertyType: string;
  icon?: string;
  onEditPress?: () => void;
  selectedPan?: string;
  badgeInfo?: IBadgeInfo;
  amenities?: IAmenitiesData[];
  onPressSteps: (index: number) => void;
  badgeStyle?: StyleProp<ViewStyle>;
  stepContainerStyle?: StyleProp<ViewStyle>;
  containerStyle?: StyleProp<ViewStyle>;
  stepIndicatorSeparatorStyle?: StyleProp<ViewStyle>;
  primaryAddressTextStyles?: ITypographyProps;
  subAddressTextStyles?: ITypographyProps;
}

export const AddressWithStepIndicator = (props: IProps): React.ReactElement => {
  const {
    steps,
    currentIndex,
    isStepDone,
    icon,
    selectedPan,
    badgeInfo,
    badgeStyle,
    amenities,
    stepContainerStyle = {},
    containerStyle = {},
    primaryAddress,
    subAddress,
    propertyType,
    onPressSteps,
    onEditPress,
    countryFlag,
    stepIndicatorSeparatorStyle,
    subAddressTextStyles,
    primaryAddressTextStyles,
  } = props;

  const { width } = useViewPort();
  const { t } = useTranslation(LocaleConstants.namespacesKey.property);
  const isMobile = useDown(deviceBreakpoint.MOBILE);
  const isTablet = useDown(deviceBreakpoint.TABLET);
  const numberOfSteps = steps.length;
  const { rental, sell, informational, blue, green, darkTint7 } = theme.colors;

  const badgeData = (): ILabelColor => {
    switch (selectedPan) {
      case TypeOfPlan.SELL:
        return {
          label: t('forSale'),
          color: sell,
        };
      case TypeOfPlan.MANAGE:
        return {
          label: t('manage'),
          color: informational,
        };
      case TypeOfPlan.RENT:
      default:
        return {
          label: t('forRental'),
          color: rental,
        };
    }
  };

  const renderIndicator = ({ item, index }: { item: string; index: number }): React.ReactElement => {
    const isCurrentStep = currentIndex === index;
    const stepDone = isStepDone[index];
    const iconColor = isCurrentStep ? blue : stepDone ? green : darkTint7;
    const labelStyle = [styles.stepLabel, isCurrentStep && styles.currentStepLabel, stepDone && styles.doneStepLabel];
    return (
      <View style={styles.indicatorView}>
        <Icon name={icons.roundFilled} size={30} color={iconColor} onPress={(): void => onPressSteps(index)} />
        <Label type="small" textType="semiBold" style={labelStyle}>
          {item}
        </Label>
      </View>
    );
  };

  const renderSeparator = (separatorData: { highlighted: boolean; leadingItem: string }): React.ReactElement => {
    let isDone = false;
    const indicatorSteps: string[] = steps;
    indicatorSteps.forEach((item, index) => {
      if (separatorData.leadingItem === item) {
        isDone = isStepDone[index];
      }
    });

    const calculateSeparatorWidth = (): number => {
      if (PlatformUtils.isMobile()) return (width - 220) / numberOfSteps;
      if (isMobile) return (width - 170) / numberOfSteps;
      if (isTablet) return (width - 230) / numberOfSteps;
      return (width - 600) / numberOfSteps;
    };

    const separatorWidth = { width: calculateSeparatorWidth() };

    return (
      <View style={[styles.separator, isDone && styles.doneSeparator, separatorWidth, stepIndicatorSeparatorStyle]} />
    );
  };

  const badge = badgeInfo ? { label: badgeInfo.title, color: badgeInfo.color } : badgeData();

  return (
    <View style={[styles.container, containerStyle]}>
      <View style={styles.topView}>
        <Label type="large" textType="regular" style={styles.propertyTypeStyle}>
          {propertyType}
        </Label>
        {icon && <Icon name={icon} size={23} color={blue} onPress={onEditPress} />}
        {(selectedPan || badgeInfo) && (
          <Badge title={badge.label.toUpperCase()} badgeColor={badge.color} badgeStyle={badgeStyle} />
        )}
      </View>
      <PropertyAddressCountry
        primaryAddress={primaryAddress}
        subAddress={subAddress}
        countryFlag={countryFlag}
        containerStyle={styles.addressView}
        primaryAddressTextStyles={primaryAddressTextStyles}
        subAddressTextStyles={subAddressTextStyles}
      />
      {amenities && (
        <PropertyAmenities
          labelStyles={subAddressTextStyles}
          containerStyle={styles.amenities}
          contentContainerStyle={styles.amenitiesContentStyle}
          data={amenities}
          direction="row"
        />
      )}
      <FlatList
        keyExtractor={(step): string => step}
        data={steps}
        renderItem={renderIndicator}
        horizontal
        showsHorizontalScrollIndicator={false}
        ItemSeparatorComponent={renderSeparator}
        style={[styles.listStyle, stepContainerStyle]}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 16,
    borderRadius: 4,
    backgroundColor: theme.colors.white,
  },
  addressView: {
    marginVertical: 6,
  },
  indicatorView: {
    alignItems: 'center',
  },
  topView: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  separator: {
    backgroundColor: theme.colors.darkTint7,
    height: 2,
    marginTop: 10,
    opacity: 0.3,
  },
  doneSeparator: {
    backgroundColor: theme.colors.green,
  },
  listStyle: {
    marginTop: 10,
    alignSelf: 'center',
  },
  stepLabel: {
    color: theme.colors.darkTint7,
  },
  currentStepLabel: {
    color: theme.colors.blue,
  },
  doneStepLabel: {
    color: theme.colors.green,
  },
  amenities: {
    marginTop: 10,
    marginBottom: 14,
    justifyContent: 'flex-start',
  },
  amenitiesContentStyle: {
    marginRight: 16,
  },
  propertyTypeStyle: {
    color: theme.colors.primaryColor,
  },
});
