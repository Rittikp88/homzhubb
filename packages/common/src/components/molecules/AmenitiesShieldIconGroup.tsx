import React, { FC } from 'react';
import { StyleSheet, View } from 'react-native';
import { PlatformUtils } from '@homzhub/common/src/utils/PlatformUtils';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';

interface IProps {
  iconSize: number;
  onBadgePress: () => void;
}

export const AmenitiesShieldIconGroup: FC<IProps> = ({ onBadgePress, iconSize }: IProps) => {
  return (
    <View style={styles.badgesContainer}>
      <Icon
        name={icons.badge}
        size={iconSize}
        color={theme.colors.warning}
        style={styles.badges}
        onPress={onBadgePress}
        testID="info"
      />
      <Icon
        name={icons.badge}
        size={iconSize}
        color={theme.colors.warning}
        style={styles.badges}
        onPress={onBadgePress}
      />
      <Icon
        name={icons.badge}
        size={iconSize}
        color={theme.colors.disabledPreference}
        style={styles.badges}
        onPress={onBadgePress}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  badgesContainer: {
    flexDirection: 'row',
  },
  badges: {
    marginHorizontal: PlatformUtils.isWeb() ? 0 : 3,
    marginLeft: PlatformUtils.isWeb() ? 4 : 0,
  },
});
