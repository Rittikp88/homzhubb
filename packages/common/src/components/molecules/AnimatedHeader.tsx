import React from 'react';
import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native';
// @ts-ignore
import ParallaxScrollView from 'react-native-parallax-scroll-view';
import { PlatformUtils } from '@homzhub/common/src/utils/PlatformUtils';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { Text, TextSizeType, WithShadowView, DetailedHeader } from '@homzhub/common/src/components';

interface IHeaderProps {
  children?: React.ReactElement;
  icon?: string;
  iconSize?: number;
  iconColor?: string;
  title?: string;
  subTitle?: string;
  subTitleColor?: string;
  subTitleType?: TextSizeType;
  linkText?: string;
  onIconPress?: () => void;
  onLinkPress?: () => void;
  headerContainerStyle?: StyleProp<ViewStyle>;
}

const PARALLAX_HEADER_HEIGHT = 200;
const STICKY_HEADER_HEIGHT = 100;

export const AnimatedHeader = (props: IHeaderProps): React.ReactElement => {
  const { children, title, subTitle, linkText, onIconPress, onLinkPress } = props;

  const stickyHeader = (): React.ReactElement => {
    return (
      <WithShadowView>
        <View key="sticky-header" style={styles.stickySection}>
          <Text type="regular" textType="semiBold" style={styles.title}>
            {title}
          </Text>
        </View>
      </WithShadowView>
    );
  };

  const fixedHeader = (): React.ReactElement => {
    return (
      <View key="fixed-header" style={styles.fixedSection}>
        <Icon name={icons.close} size={22} color={theme.colors.darkTint4} onPress={onIconPress} />
      </View>
    );
  };

  return (
    <ParallaxScrollView
      backgroundColor={theme.colors.white}
      stickyHeaderHeight={STICKY_HEADER_HEIGHT}
      parallaxHeaderHeight={PARALLAX_HEADER_HEIGHT}
      backgroundSpeed={10}
      renderForeground={(): React.ReactElement => (
        <DetailedHeader title={title} subTitle={subTitle} linkText={linkText} onLinkPress={onLinkPress} />
      )}
      renderStickyHeader={(): React.ReactElement => stickyHeader()}
      renderFixedHeader={(): React.ReactElement => fixedHeader()}
    >
      {children}
    </ParallaxScrollView>
  );
};

const styles = StyleSheet.create({
  stickySection: {
    height: PlatformUtils.isIOS() ? 95 : 80,
    paddingTop: PlatformUtils.isIOS() ? 55 : 30,
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderBottomColor: theme.colors.disabled,
  },
  title: {
    color: theme.colors.dark,
    flex: 1,
    textAlign: 'center',
  },
  fixedSection: {
    position: 'absolute',
    bottom: PlatformUtils.isIOS() ? 20 : 40,
    left: 20,
  },
});
