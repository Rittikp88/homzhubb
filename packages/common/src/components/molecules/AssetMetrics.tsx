import React, { useState } from 'react';
import { StyleProp, StyleSheet, TouchableOpacity, ViewStyle } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import { theme } from '@homzhub/common/src/styles/theme';
import { Text } from '@homzhub/common/src/components/atoms/Text';
import { PricePerUnit } from '@homzhub/common/src/components/atoms/PricePerUnit';

interface IProps {
  header: string;
  value: string | number;
  currency?: string;
  cardStyle?: StyleProp<ViewStyle>;
  angle: number;
  location: number[];
  colorA: string;
  colorB: string;
  showPlusIcon?: boolean;
  testID?: string;
}

const AssetMetrics = (props: IProps): React.ReactElement => {
  const { header, value, angle, colorA, colorB, location, showPlusIcon = false, currency, cardStyle, testID } = props;

  const [selected, onSelect] = useState(false);
  const gradient = [colorA || theme.colors.gradientK, colorB || theme.colors.white];

  const handlePress = (): void => {
    onSelect(!selected);
  };

  return (
    <TouchableOpacity onPress={handlePress} testID={testID}>
      <LinearGradient
        useAngle
        angle={angle}
        colors={gradient}
        locations={location}
        style={[
          styles.container,
          cardStyle,
          showPlusIcon && selected ? styles.selectedContainer : styles.containerWithoutGradient,
        ]}
      >
        <Text
          type="small"
          textType="semiBold"
          style={[styles.metrics, !showPlusIcon ? styles.textWithGradient : styles.textWithoutGradient]}
        >
          {header}
        </Text>
        {currency ? (
          <PricePerUnit priceTransformation={false} currency={currency} price={value as number} />
        ) : (
          <Text
            type="large"
            textType="semiBold"
            style={[styles.metrics, !showPlusIcon ? styles.textWithGradient : styles.valueWithoutGradient]}
          >
            {value}
          </Text>
        )}
      </LinearGradient>
    </TouchableOpacity>
  );
};

export { AssetMetrics };

const styles = StyleSheet.create({
  container: {
    margin: 10,
    borderRadius: 8,
    paddingVertical: 10,
    paddingHorizontal: 5,
    minWidth: (theme.viewport.width - 90) / 3,
    shadowColor: theme.colors.shadow,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  containerWithoutGradient: {
    borderColor: theme.colors.lightBlue,
    borderWidth: 1,
  },
  selectedContainer: {
    borderColor: theme.colors.blue,
    borderWidth: 1,
  },
  metrics: {
    textAlign: 'center',
    marginVertical: 5,
  },
  textWithGradient: {
    color: theme.colors.white,
  },
  textWithoutGradient: {
    color: theme.colors.darkTint4,
  },
  valueWithoutGradient: {
    color: theme.colors.darkTint1,
  },
});
