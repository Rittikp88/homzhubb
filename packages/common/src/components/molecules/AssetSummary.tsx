import React from 'react';
import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native';
import { useTranslation } from 'react-i18next';
import LinearGradient from 'react-native-linear-gradient';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { Text } from '@homzhub/common/src/components/atoms/Text';
import { Divider } from '@homzhub/common/src/components/atoms/Divider';

interface IProps {
  notification?: number;
  serviceTickets?: number;
  dues?: number;
  containerStyle?: StyleProp<ViewStyle>;
}

const AssetSummary = (props: IProps): React.ReactElement => {
  const { t } = useTranslation(LocaleConstants.namespacesKey.assetDashboard);
  const { notification = 0, serviceTickets = 0, dues = 0, containerStyle } = props;

  return (
    <LinearGradient
      useAngle
      angle={180}
      colors={[theme.colors.white, theme.colors.background]}
      locations={[0, 1]}
      style={[styles.container, containerStyle]}
    >
      <View style={styles.summary}>
        <Icon name={icons.bell} color={theme.colors.green} size={40} />
        <Text type="small" textType="semiBold" style={styles.notification}>
          {t('notification')}
        </Text>
        <Text type="large" textType="bold" style={styles.notification}>
          {notification}
        </Text>
      </View>
      <Divider />
      <View style={styles.summary}>
        <Icon name={icons.headPhone} color={theme.colors.warning} size={40} />
        <Text type="small" textType="semiBold" style={styles.serviceTickets}>
          {t('serviceTickets')}
        </Text>
        <Text type="large" textType="bold" style={styles.serviceTickets}>
          {serviceTickets}
        </Text>
      </View>
      <Divider />
      <View style={styles.summary}>
        <Icon name={icons.decrease} color={theme.colors.danger} size={40} />
        <Text type="small" textType="semiBold" style={styles.dues}>
          {t('dues')}
        </Text>
        <Text type="large" textType="bold" style={styles.dues}>
          {dues}
        </Text>
      </View>
    </LinearGradient>
  );
};

export { AssetSummary };

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    backgroundColor: theme.colors.white,
    padding: 10,
    borderRadius: 10,
    justifyContent: 'space-around',
  },
  summary: {
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  notification: {
    color: theme.colors.green,
    marginVertical: 2,
  },
  serviceTickets: {
    color: theme.colors.warning,
    marginVertical: 2,
  },
  dues: {
    color: theme.colors.danger,
    marginVertical: 2,
  },
});
