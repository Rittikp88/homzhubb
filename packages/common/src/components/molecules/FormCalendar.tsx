import React, { Component } from 'react';
import { View, StyleSheet, StyleProp, ViewStyle, TouchableOpacity, TextStyle } from 'react-native';
import { FormikProps } from 'formik';
import moment from 'moment';
import { withTranslation, WithTranslation } from 'react-i18next';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { theme } from '@homzhub/common/src/styles/theme';
import { Text, Label, TextFieldType, TextSizeType, FontWeightType } from '@homzhub/common/src/components/atoms/Text';
import { BottomSheet } from '@homzhub/mobile/src/components/molecules/BottomSheet';
import { CalendarComponent } from '@homzhub/common/src/components/atoms/CalendarComponent';

interface IFormCalendarProps extends WithTranslation {
  name: string;
  formProps?: FormikProps<any>;
  selectedValue?: string;
  label?: string;
  iconColor?: string;
  placeHolder?: string;
  allowPastDates?: boolean;
  containerStyle?: StyleProp<ViewStyle>;
  placeHolderStyle?: StyleProp<TextStyle>;
  textType?: TextFieldType;
  textSize?: TextSizeType;
  fontType?: FontWeightType;
  maxDate?: string;
  bubbleSelectedDate?: (day: string) => void;
}
interface IFormCalendarState {
  isCalendarVisible: boolean;
}
class FormCalendar extends Component<IFormCalendarProps, IFormCalendarState> {
  public state = {
    isCalendarVisible: false,
  };

  public render(): React.ReactNode {
    const {
      t,
      name,
      formProps,
      selectedValue,
      containerStyle = {},
      label,
      placeHolder,
      allowPastDates,
      textType,
      iconColor,
      textSize = 'regular',
      fontType = 'regular',
      placeHolderStyle = {},
      maxDate,
    } = this.props;
    const { isCalendarVisible } = this.state;
    const availableDate = (): string => {
      if (selectedValue) {
        return selectedValue;
      }
      return formProps?.values[name] === moment().format('YYYY-MM-DD') ? 'Today' : formProps?.values[name];
    };
    const labelStyles = { ...theme.form.formLabel };
    let TextField = Text;
    if (textType === 'label') {
      TextField = Label;
    }
    return (
      <View style={containerStyle}>
        <TextField type={textSize} textType={fontType} style={labelStyles}>
          {label || t('common:availableFrom')}
        </TextField>
        <TouchableOpacity testID="toCalenderInput" style={styles.dateView} onPress={this.onCalendarOpen}>
          <View style={styles.dateLeft}>
            <Icon name={icons.calendar} color={iconColor || theme.colors.darkTint5} size={18} />
            <Text type="small" textType="regular" style={[styles.dateText, selectedValue === '' && placeHolderStyle]}>
              {availableDate() || placeHolder}
            </Text>
          </View>
          <Icon name={icons.downArrowFilled} color={theme.colors.darkTint7} size={16} />
        </TouchableOpacity>
        <BottomSheet
          visible={isCalendarVisible}
          onCloseSheet={this.onCalendarClose}
          headerTitle={t('common:availableFrom')}
          isShadowView
          sheetHeight={580}
        >
          <CalendarComponent
            allowPastDates={allowPastDates}
            maxDate={maxDate}
            onSelect={this.onDateSelected}
            selectedDate={selectedValue ?? formProps?.values[name]}
          />
        </BottomSheet>
      </View>
    );
  }

  private onDateSelected = (day: string): void => {
    const { name, formProps, bubbleSelectedDate } = this.props;
    this.setState({ isCalendarVisible: false });
    if (bubbleSelectedDate) {
      bubbleSelectedDate(day);
    } else {
      formProps?.setFieldValue(name, day);
      formProps?.setFieldTouched(name);
    }
  };

  private onCalendarOpen = (): void => {
    const { isCalendarVisible } = this.state;
    this.setState({ isCalendarVisible: !isCalendarVisible });
  };

  private onCalendarClose = (): void => {
    this.setState({ isCalendarVisible: false });
  };
}
const styles = StyleSheet.create({
  dateView: {
    borderWidth: 1,
    borderColor: theme.colors.disabled,
    flexDirection: 'row',
    paddingVertical: 12,
    paddingHorizontal: 20,
    alignItems: 'center',
    justifyContent: 'space-between',
    borderRadius: 4,
    marginTop: 6,
  },
  dateLeft: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  dateText: {
    marginLeft: 16,
    color: theme.colors.darkTint1,
  },
});
const HOC = withTranslation()(FormCalendar);
export { HOC as FormCalendar };
