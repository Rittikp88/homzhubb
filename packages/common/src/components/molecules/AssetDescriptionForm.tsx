import React from 'react';
import { StyleSheet, View, ViewStyle } from 'react-native';
import { useTranslation } from 'react-i18next';
import { FormikProps, FormikValues } from 'formik';
import { useDown } from '@homzhub/common/src/utils/MediaQueryUtils';
import { theme } from '@homzhub/common/src/styles/theme';
import { FormDropdown } from '@homzhub/common/src/components/molecules/FormDropdown';
import { FormTextInput } from '@homzhub/common/src/components/molecules/FormTextInput';
import { FormCounter } from '@homzhub/common/src/components/molecules/FormCounter';
import { FormCalendar } from '@homzhub/common/src/components/molecules/FormCalendar';
import { AssetDescriptionDropdownValues } from '@homzhub/common/src/domain/models/AssetDescriptionForm';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { deviceBreakpoint } from '@homzhub/common/src/constants/DeviceBreakpoints';

interface IProps {
  formProps: FormikProps<FormikValues>;
  dropDownOptions: AssetDescriptionDropdownValues;
}

const AssetDescriptionForm = ({ formProps, dropDownOptions }: IProps): React.ReactElement => {
  const [t] = useTranslation(LocaleConstants.namespacesKey.property);
  const isMobile = useDown(deviceBreakpoint.MOBILE);
  const styles = formStyles(isMobile);
  return (
    <View style={styles.formFieldsContainer}>
      <View style={styles.formFieldsContainer}>
        <View style={styles.contentView}>
          <View style={styles.subContentView}>
            <FormTextInput
              style={styles.inputFieldStyle}
              name="carpetArea"
              label={t('propertySearch:carpetArea')}
              maxLength={10}
              numberOfLines={1}
              inputType="number"
              placeholder={t('propertySearch:carpetAreaPlaceholder')}
              formProps={formProps}
            />
          </View>
          <View style={styles.flexOne}>
            <FormDropdown
              name="areaUnit"
              label={t('areaUnit')}
              options={dropDownOptions.areaUnitDropdownValues}
              placeholder={t('selectAreaUnit')}
              formProps={formProps}
            />
          </View>
        </View>
        <View style={styles.formFields}>
          <FormDropdown
            label={t('facingText')}
            name="facing"
            options={dropDownOptions.facing}
            placeholder={t('propertySearch:selectFacing')}
            formProps={formProps}
          />
        </View>
        <View style={styles.formFields}>
          <FormDropdown
            label={t('flooringType')}
            name="flooringType"
            options={dropDownOptions.typeOfFlooring}
            placeholder={t('selectFlooringType')}
            formProps={formProps}
          />
        </View>
        <View style={styles.formFields}>
          <FormCalendar
            formProps={formProps}
            label={t('yearOfPossession')}
            name="yearOfConstruction"
            allowPastDates
            isYearView
            calendarTitle={t('yearOfPossession')}
            placeHolder={t('assetFinancial:addDatePlaceholder')}
            placeHolderStyle={styles.placeHolderStyle}
            dateStyle={styles.dateStyle}
            dateContainerStyle={styles.dateContainerStyle}
            textType="label"
          />
        </View>
      </View>
      <FormCounter
        containerStyles={styles.formCounter}
        name="totalFloors"
        maxCount={1000}
        label={t('numberOfFloorsText')}
        formProps={formProps}
      />
      <FormCounter
        containerStyles={styles.formCounter}
        name="onFloorNumber"
        label={t('onFloorText')}
        formProps={formProps}
        maxCount={formProps.values.totalFloors}
      />
    </View>
  );
};

interface IFormStyles {
  formContainer: ViewStyle;
  contentView: ViewStyle;
  subContentView: ViewStyle;
  formFields: ViewStyle;
  formFieldsContainer: ViewStyle;
  formCounter: ViewStyle;
  flexOne: ViewStyle;
  inputFieldStyle: ViewStyle;
  dateStyle: ViewStyle;
  dateContainerStyle: ViewStyle;
  placeHolderStyle: ViewStyle;
}

const formStyles = (isMobile: boolean): StyleSheet.NamedStyles<IFormStyles> =>
  StyleSheet.create<IFormStyles>({
    formFieldsContainer: {
      width: '100%',
      flexDirection: 'row',
      flexWrap: 'wrap',
    },
    formContainer: {
      flexDirection: isMobile ? undefined : 'row',
      flexWrap: isMobile ? undefined : 'wrap',
      backgroundColor: theme.colors.white,
      paddingBottom: 24,
    },
    contentView: {
      flexShrink: isMobile ? 1 : undefined,
      flexDirection: 'row',
      alignItems: 'center',
    },
    subContentView: {
      flex: 1,
      marginRight: 16,
    },
    formFields: {
      width: isMobile ? '100%' : undefined,
      marginLeft: isMobile ? 0 : 24,
      justifyContent: 'center',
    },
    formCounter: {
      marginTop: 24,
      marginRight: isMobile ? 0 : 24,
    },
    flexOne: {
      flex: 1,
    },
    inputFieldStyle: {
      height: 40,
    },
    dateStyle: {
      marginLeft: 0,
    },
    dateContainerStyle: {
      paddingVertical: 8,
      marginTop: 0,
    },
    placeHolderStyle: {
      color: theme.colors.darkTint8,
    },
  });

const memoizedComponent = React.memo(AssetDescriptionForm);
export { memoizedComponent as AssetDescriptionForm };
