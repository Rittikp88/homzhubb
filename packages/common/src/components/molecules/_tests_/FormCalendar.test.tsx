import React from 'react';
import moment from 'moment';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import { BottomSheet } from '@homzhub/mobile/src/components/molecules/BottomSheet';
import { CalendarComponent } from '@homzhub/common/src/components/atoms/CalendarComponent';
import { FormCalendar } from '@homzhub/common/src/components/molecules/FormCalendar';

jest.mock('@homzhub/common/src/services/storage/StorageService', () => 'StorageService');
jest.mock('@react-native-community/google-signin', () => {});

describe('Test cases for FormCalendar', () => {
  const formProps = {
    values: {
      test: '2020-11-01',
    },
    touched: {
      name: 'test',
    },
    setFieldValue: jest.fn(),
    setFieldTouched: jest.fn(),
  };
  // @ts-ignore
  const wrapper: ShallowWrapper = shallow(<FormCalendar name="test" formProps={formProps} />).dive();

  it('should render snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render snapshot when initial date today', () => {
    wrapper.setProps({
      formProps: {
        ...formProps,
        values: {
          test: moment().format('YYYY-MM-DD'),
        },
      },
    });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render snapshot when calendar is open', () => {
    expect(wrapper.state('isCalendarVisible')).toEqual(false);
    wrapper.find("[testID='toCalenderInput']").simulate('press');
    expect(wrapper.state('isCalendarVisible')).toEqual(true);
  });

  it('should close bottom-sheet on close button press of bottom-sheet', () => {
    expect(wrapper.setState({ isCalendarVisible: true }));
    // @ts-ignore
    wrapper.find(BottomSheet).prop('onCloseSheet')();
    expect(wrapper.state('isCalendarVisible')).toEqual(false);
  });

  it('should set the form field when user selects a date', () => {
    const selectedDate = '2020-11-01';
    expect(wrapper.setState({ isCalendarVisible: true }));
    wrapper.find(CalendarComponent).prop('onSelect')(selectedDate);
    expect(wrapper.state('isCalendarVisible')).toEqual(false);
    expect(formProps.setFieldTouched).toHaveBeenCalledWith('test');
    expect(formProps.setFieldValue).toHaveBeenCalledWith('test', selectedDate);
  });
});
