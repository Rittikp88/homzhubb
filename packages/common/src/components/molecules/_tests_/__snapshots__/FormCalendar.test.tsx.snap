// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases for FormCalendar should render snapshot 1`] = `
<View
  style={Object {}}
>
  <Text
    style={
      Object {
        "color": "#33495E",
        "marginBottom": 6,
        "marginTop": 16,
      }
    }
    textType="regular"
    type="regular"
  >
    common:availableFrom
  </Text>
  <ForwardRef
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#C9D6DF",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 6,
        "paddingHorizontal": 20,
        "paddingVertical": 12,
      }
    }
    testID="toCalenderInput"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="#667686"
        name="calendar"
        size={18}
      />
      <Text
        style={
          Array [
            Object {
              "color": "#001B36",
              "marginLeft": 16,
            },
            false,
          ]
        }
        textType="regular"
        type="small"
      >
        2020-11-01
      </Text>
    </View>
    <Icon
      allowFontScaling={false}
      color="#99A4AF"
      name="down-arrow-filled"
      size={16}
    />
  </ForwardRef>
  <BottomSheet
    headerTitle="common:availableFrom"
    isShadowView={true}
    onCloseSheet={[Function]}
    sheetHeight={580}
    visible={false}
  >
    <CalendarComponent
      onSelect={[Function]}
      selectedDate="2020-11-01"
    />
  </BottomSheet>
</View>
`;

exports[`Test cases for FormCalendar should render snapshot when initial date today 1`] = `
<View
  style={Object {}}
>
  <Text
    style={
      Object {
        "color": "#33495E",
        "marginBottom": 6,
        "marginTop": 16,
      }
    }
    textType="regular"
    type="regular"
  >
    common:availableFrom
  </Text>
  <ForwardRef
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#C9D6DF",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 6,
        "paddingHorizontal": 20,
        "paddingVertical": 12,
      }
    }
    testID="toCalenderInput"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="#667686"
        name="calendar"
        size={18}
      />
      <Text
        style={
          Array [
            Object {
              "color": "#001B36",
              "marginLeft": 16,
            },
            false,
          ]
        }
        textType="regular"
        type="small"
      >
        Today
      </Text>
    </View>
    <Icon
      allowFontScaling={false}
      color="#99A4AF"
      name="down-arrow-filled"
      size={16}
    />
  </ForwardRef>
  <BottomSheet
    headerTitle="common:availableFrom"
    isShadowView={true}
    onCloseSheet={[Function]}
    sheetHeight={580}
    visible={false}
  >
    <CalendarComponent
      onSelect={[Function]}
      selectedDate="2020-10-15"
    />
  </BottomSheet>
</View>
`;
